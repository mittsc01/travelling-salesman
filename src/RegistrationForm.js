import React from 'react'
import AuthApiService from './services/auth-api-service'





export default class RegistrationForm extends React.Component {
    
    
    
      state = { error: null }
    
      handleSubmit = ev => {
        ev.preventDefault()
        const { full_name, user_name, password,repeat_password } = ev.target
        
        
        AuthApiService.postUser({
          user_name: user_name.value,
          password: password.value,
          full_name: full_name.value,
          repeat_password: repeat_password.value
          
        })
          .then(user => {
            full_name.value = ''
            
            user_name.value = ''
            password.value = ''
            repeat_password.value = ''
            this.props.history.push('/login')
            
          })
          .catch(res => {
            this.setState({ error: res.error })
          })
      }
    render(){
        return (
        <form onSubmit={this.handleSubmit} className="registration-form">
            <label htmlFor="full_name">Name</label>
            <input name="full_name"></input>
            <label htmlFor="user_name">Username</label>
            <input type="text" name="user_name"/>
            <label htmlFor="password">Password</label>
            <input type="password" name="password"/>
            <label htmlFor="repeat_password">Repeat Password</label>
            <input type="password" name="repeat_password"/>
            <button type="submit">Register</button>
        </form>
            )
    }
}