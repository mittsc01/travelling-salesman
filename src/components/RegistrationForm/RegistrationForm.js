import React from 'react'
import AuthApiService from '../../services/auth-api-service'





export default class RegistrationForm extends React.Component {
    
    
    
      state = { error: null }
    
      handleSubmit = ev => {
        ev.preventDefault()
        const { full_name, user_name, password,repeat_password } = ev.target
        
        if (password.value === repeat_password.value){
          AuthApiService.postUser({
            user_name: user_name.value,
            password: password.value,
            full_name: full_name.value,
            
            
          })
            .then(user => {
              full_name.value = ''
              
              user_name.value = ''
              password.value = ''
              repeat_password.value = ''
              this.props.history.push('/login')
              
            })
            .catch(res => {
              this.setState({ error: res.error })
            })
        }
        else {
          
          alert('Passwords do not match. Please try again.')
          password.value = ''
          repeat_password.value = ''
        }
        
        
      }
    render(){
        return (
        <form onSubmit={this.handleSubmit} className="registration-form">
            <label htmlFor="full_name">Name</label>
            <input name="full_name"></input>
            <label htmlFor="user_name">Username</label>
            <input type="text" name="user_name"/>
            <label htmlFor="password">Password</label>
            <input type="password" name="password"/>
            <label htmlFor="repeat_password">Repeat Password</label>
            <input type="password" name="repeat_password"/>
            <button type="submit">Register</button>
        </form>
            )
    }
}