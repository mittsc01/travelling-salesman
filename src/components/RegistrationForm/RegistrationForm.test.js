import RegistrationForm from './RegistrationForm';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');

  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<RegistrationForm />, div);

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<RegistrationForm />)
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });