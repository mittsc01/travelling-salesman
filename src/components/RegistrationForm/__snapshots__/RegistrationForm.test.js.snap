// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<form
  className="registration-form"
  onSubmit={[Function]}
>
  <label
    htmlFor="full_name"
  >
    Name
  </label>
  <input
    name="full_name"
  />
  <label
    htmlFor="user_name"
  >
    Username
  </label>
  <input
    name="user_name"
    type="text"
  />
  <label
    htmlFor="password"
  >
    Password
  </label>
  <input
    name="password"
    type="password"
  />
  <label
    htmlFor="repeat_password"
  >
    Repeat Password
  </label>
  <input
    name="repeat_password"
    type="password"
  />
  <button
    type="submit"
  >
    Register
  </button>
</form>
`;
